feat(security): update reCAPTCHA integration to v3 and add detailed v…

BlackRomeoIlch opened the pull request:
…alidation

- Implemented reCAPTCHA v3 server-side verification with score and action handling
- Added detailed validation results (score, action, error_codes)
- Updated admin settings to support v3 keys and threshold
- Added translations (de/en)

# Description
Update der reCAPTCHA-Integration auf v3 inkl. serverseitiger Detail-Validierung (Score, Action, Error-Codes).
Motivation: Spam reduzieren, an aktuellen Google-Flow anpassen, Sicherheit erhöhen.
Änderungen:
- Neue Methode `validateDetailed(token, action, scoreThreshold)` mit Score/Action/Fehlercodes
- Admin-Settings für v3: Site Key, Secret Key, `score_threshold`
- Action-Namen pro Formular (z. B. `contact_submit`, `register_submit`)
- Logs mit Score und Error-Codes, bereinigte Fehlermeldungen
- Übersetzungen (de/en) und Admin-View aktualisiert
Migration:
- v3-Keys im Admin hinterlegen (v2-Keys sind nicht kompatibel)
- Optional: Cache leeren nach Deployment
Abhängigkeiten: keine externen Pakete, nur neue Konfig-Einträge.

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# This PR has been tested in the following browsers:
- [x] Chrome
- [x] Firefox
- [ ] Opera
- [ ] Edge

BlackRomeoIlch committed:
feat(security): update reCAPTCHA integration to v3 and add detailed validation

- Implemented reCAPTCHA v3 server-side verification with score and action handling
- Added detailed validation results (score, action, error_codes)
- Updated admin settings to support v3 keys and threshold
- Added translations (de/en)

PrivatePHPCoder commented:
Laut der Google Doku sollten noch einige Sicherheitslücken geschlossen werden. XSS, Validierung,  Escapen etc....

BlackRomeoIlch committed:
Fix: Recaptcha V3 XSS, Validierung, Escapen

Die Implementierung entspricht den Best Practices:
XSS-Abwehr durch Escaping und Client-seitige Prüfen
Server-seitige Input-Validierung und Sanitization
Sichere SSL/TLS-Konfiguration
Hostname-Verifikation
Klare Layer: Validator → GoogleCaptcha
Robust gegenüber Manipulationen

BlackRomeoIlch committed:
Fix:ReCaptcha SettingsController Captcha_Score View

Der captcha_score wird mit Standardwert 0.5 an die View übergeben; die View zeigt ihn im Captcha-Bereich an.

BlackRomeoIlch commented:
> Habe es nur überflogen...aber im SettingsController muss der captcha_Score noch an die View übergeben werden. In der View wird im Input darauf zugegriffen....

Hey Danke für den Hinweis daran hatte ich Gar nicht gedacht Sorry.............
Wen noch weiteres Findest immer gerne raus damit........

blackcoder87 commented on application/libraries/Captcha/GoogleCaptcha.php:
```diff
@@ -205,27 +295,107 @@ public function getCaptcha(\Ilch\View $view, string $saveKey = 'saveButton', ?st
+                        
+                        const response = grecaptcha.getResponse();
+                        if (!response) {
+                            alert("Bitte lösen Sie das reCAPTCHA");
```
This message is visible to the user and therefore should have a translation.
